src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -288,6 +296,30 @@ async def _add_objects_to_archive(
     return pack_result
 
 
+def _maybe_is_golang_assembly(data: bytes) -> bool:
+    return (
+        data.startswith(b"TEXT")
+        or b"\nTEXT" in data
```

Comment:
These checks anywhere in the body are pretty broad... a comment explaining why we check "anywhere" might be good. To skip a header of some sort?

Also, you might consider switching these to a regex to match in one pass (since this could be applied to a lot of files).

Reply:
This is a transcription of the same check in the `go` toolchain.

Reply:
Added a comment explaining what this function does.